sdk/freestanding-execution-engine/src/main.rs

```diff
@@ -235,13 +253,29 @@ fn parse_command_line() -> Result<CommandLineOptions, Box<dyn Error>> {
         }
     };
 
+    let environment_variables = match matches.values_of("env") {
+        None => Vec::new(),
+        Some(x) =>
+            x.map(|e| {
+                let n = e.find("=").unwrap();
+                (e[0..n].to_string(), e[n + 1..].to_string())
+            })
+            .collect(),
+    };
+    let program_arguments = match matches.values_of("arg") {
+        None => Vec::new(),
+        Some(x) => x.map(|e| e.to_string()).collect(),
+    };
+
     Ok(CommandLineOptions {
         data_sources,
         binary,
         execution_strategy,
         dump_stdout,
         dump_stderr,
         enable_clock,
+        environment_variables,
+        program_arguments,
```

Comment:
How are the arguments and environment variables passed to the program? `main()` doesn't seem to be using `environment_variables` and `program_arguments`.

Reply:
Is that clearer now with the single commit?

Reply:
Oh I hadn't noticed the syscalls were already implemented, `environ_get()` and `args_get()`

Reply:
Yes, this patch is small because a lot of the stuff was there already. Apart from `sdk/freestanding-execution-engine` it's 11 lines!